File: sistema veterinario/Tratamiento.java
public class Tratamiento
{
    private String descripcion;
    private Animal animal;

    public Tratamiento(String descripcion,
                       Animal animal)
    {
        this.descripcion = descripcion;
        this.animal = animal;
    }

    public void aplicar()
    {
        System.out.println("Tratamiento: "
                           + descripcion);

        animal.aplicarTratamiento();
    }
}

File: Logica/Tratamiento.cs
Tratamiento.cs                                               namespace Sistema_Veterinario.Logica
{
    public class Tratamiento
    {
        public string Descripcion { get; set; }

        public Tratamiento(string descripcion)
        {
            Descripcion = descripcion;
        }

        public void Mostrar()
        {
            Console.WriteLine(
                "Tratamiento: " +
                Descripcion);
        }
    }
}